File: application/core/src/main/java/br/com/beblue/bluerecords/core/repositorio/DiscoRepositorio.java
package br.com.beblue.bluerecords.core.repositorio;

import br.com.beblue.bluerecords.core.command.CadastraDiscoCommand;
import br.com.beblue.bluerecords.core.command.ConsultaDiscoCommand;
import br.com.beblue.bluerecords.core.entitidade.Disco;
import br.com.beblue.bluerecords.core.paginacao.Paginacao;

public interface DiscoRepositorio {
    Disco cadastrar(CadastraDiscoCommand command);
    Boolean existeDiscoCadastrado();
    Paginacao<Disco> consultarDiscos(ConsultaDiscoCommand command);
    Disco consultar(Integer id);
}

File: application/core/src/main/kotlin/br/com/beblue/bluerecords/core/repositorio/DiscoRepositorio.kt
package br.com.beblue.bluerecords.core.repositorio

import br.com.beblue.bluerecords.core.entity.Disco
import br.com.beblue.bluerecords.core.usecase.disco.command.CadastrarDiscoCommand

interface DiscoRepositorio {
    fun cadastrar(command: CadastrarDiscoCommand): Disco
    fun existeDiscoCadastrado(): Boolean
}